spark3-extensions/src/main/scala/org/apache/spark/sql/execution/datasources/v2/ExtendedDataSourceV2Strategy.scala

```diff
@@ -20,14 +20,21 @@
 package org.apache.spark.sql.execution.datasources.v2
 
 import org.apache.spark.sql.{AnalysisException, Strategy}
-import org.apache.spark.sql.catalyst.plans.logical.{CallStatement, LogicalPlan}
+import org.apache.spark.sql.catalyst.CatalystTypeConverters
+import org.apache.spark.sql.catalyst.expressions.{Expression, Literal}
+import org.apache.spark.sql.catalyst.plans.logical.{Call, LogicalPlan}
 import org.apache.spark.sql.execution.SparkPlan
 
 object ExtendedDataSourceV2Strategy extends Strategy {
 
   override def apply(plan: LogicalPlan): Seq[SparkPlan] = plan match {
-    case _: CallStatement =>
-      throw new AnalysisException("CALL statements are not currently supported")
+    case c @ Call(procedure, args) =>
+      CallExec(c.output, procedure.methodHandle, args.map(toScalaValue)) :: Nil
```

Comment:
Should this be in `ExtendedDataSourceV2Strategy`? It doesn't seem related to DSv2.

Reply:
@rdblue, I think this is debatable. It does depend on the catalog API part of DS V2 so I've put it here. Any suggestions?